src/redis-cli.c

```diff
@@ -1930,22 +1960,28 @@ static int confirmWithYes(char *msg, int ignore_force) {
     return (nread != 0 && !strcmp("yes", buf));
 }
 
-/* Turn the plain C strings into Sds strings */
-static char **convertToSds(int count, char** args) {
-    int j;
-    char **sds = zmalloc(sizeof(char*)*count);
-
-    for(j = 0; j < count; j++)
-        sds[j] = sdsnew(args[j]);
-
-    return sds;
-}
+/* Create an sds array from argv, either as-is or by dequoting every
+ * element. When quoted is non-zero, may return a NULL to indicate an
+ * invalid quoted string.
+ */
+static sds *getSdsArrayFromArgv(int argc, char **argv, int quoted) {
+    sds *res = sds_malloc(sizeof(sds) * argc);
```

Comment:
i don't like the use of `sds_malloc`, is this done because you wanna use `sdsfreesplitres` to release it?
maybe it would be nicer to add a dedicated release function (and mention it in the top comment of this function)

Reply:
Me neither, but also not a fan of duplicating this and I'm expecting sds to go away from hiredis at some point so we won't need to deal with that.